src/lib/InstantSearch.js

```diff
@@ -52,22 +89,28 @@ class InstantSearch extends EventEmitter {
     searchFunction,
     createAlgoliaClient = defaultCreateAlgoliaClient,
     stalledSearchDelay = 200,
+    searchClient = {},
   }) {
     super();
-    if (appId === null || apiKey === null || indexName === null) {
-      const usage = `
-Usage: instantsearch({
-  appId: 'my_application_id',
-  apiKey: 'my_search_api_key',
-  indexName: 'my_index_name'
-});`;
-      throw new Error(usage);
-    }
 
-    const client = createAlgoliaClient(algoliasearch, appId, apiKey);
-    client.addAlgoliaAgent(`instantsearch.js ${version}`);
+    // eslint-disable-next-line prefer-rest-params
+    const [THROWAWAY_PARAMS] = arguments;
```

Comment:
this line is a bit odd, what about giving the args to this function this name, and destructure after this line?

Reply:
It would indeed be cleaner. I wanted to change the fewest possible number of lines since this is meant to be thrown away soon.

Is it still worth it?

Reply:
Maybe leave the code as-is but add a comment that this is the first argument, which will be removed? Not totally sure what to do here